src/Relude/List.hs

```diff
@@ -61,6 +62,21 @@ infix 9 !!?
     go _ []     = Nothing
 {-# INLINE (!!?) #-}
 
+{- | Partitions a list based on the result of function which produces an Either value. List of all elements producing Left are extracted, in order, to the first element of the output tuple. Similarly, a list of all elements producing Right are extracted to the second element of output.
+
+>>> :{
+ foo x
+   | even x = Left x
+   | otherwise = Right x
```

Comment:
Let's change example a little bit. To understand better, why `partitionEithers` might be useful, it's better to give an example where return types are different. For example, I propose to implement the following function:

```haskell
divideEvenOrShow :: Int -> Either Int String
divideEvenOrShow n
    | even n = Left $ n `div` 2
    | otherwise = Right $ "Odd: " <> show n
```